services/harmony/app/models/services/base-service.ts

```diff
@@ -510,15 +510,18 @@ export default abstract class BaseService<ServiceParamType> {
 
           if (step.extra_args) {
             this.operation.extraArgs = step.extra_args;
-          } else {
-            // clear out extraArgs used by other steps
+          } else if (i > 1) {
```

Comment:
We shouldn't base this on the step number - we should change the check to explicitly list which extra args you do not want to remove.

I think rather than doing the check here it should be in the `removeExtraArgs` function.

However, I'm also realizing there's an issue with the download service - it uses extraArgs for the queryCmr step so line 512 is going to replace the granule validation extra args with the extraArgs from services.yml.

Reply:
Good point with the download service. I changed code to make download service work with granule validation when forceAsync=true.